FB to discord messages not working since npm update

The bot crashed out on me last night, and there was a message that npm needed to be updated, which I did.  After that npm start wouldnt work, so I used npm install and then npm start and it seemed to start ok, however it only lasted a few seconds before crashing out.    

I noticed a new version had been uploaded to github, so I have downloaded this, copied over the required information to the config file, ran npm install / npm start again and the bot ran ok.   I made a message from the discord side to say the bot had been restarted and this appeared in the FB messenger with no problem.   Someone then replied on FB, and the bot crashed out.

This is the error report  (I have replaced my folder names with ..\ to shorten the text

```
ERR! listen TypeError: config.facebook.client.getThreadInfoGraphQL is not a function
ERR! listen     at module.exports.opts (..\Facebook-to-discord\lib\listeners\facebook.js:19:26)
ERR! listen     at config.facebook.client.listen (..\Facebook-to-discord\index.js:12:51)
ERR! listen     at resolveAttachmentUrl (..\Facebook-to-discord\node_modules\facebook-chat-api\src\listen.js:262:29)
ERR! listen     at parsePackets (..\Facebook-to-discord\node_modules\facebook-chat-api\src\listen.js:281:23)
ERR! listen     at Array.forEach (<anonymous>)
ERR! listen     at ..\Facebook-to-discord\node_modules\facebook-chat-api\src\listen.js:151:14
ERR! listen     at tryCatcher (..\Facebook-to-discord\node_modules\bluebird\js\main\util.js:26:23)
ERR! listen     at Promise._settlePromiseFromHandler (..\Facebook-to-discord\node_modules\bluebird\js\main\promise.js:510:31)
ERR! listen     at Promise._settlePromiseAt (..\Facebook-to-discord\node_modules\bluebird\js\main\promise.js:584:18)
ERR! listen     at Promise._settlePromises (..\Facebook-to-discord\node_modules\bluebird\js\main\promise.js:700:14)
ERR! listen     at Async._drainQueue (..\Facebook-to-discord\node_modules\bluebird\js\main\async.js:123:16)
ERR! listen     at Async._drainQueues (..\Facebook-to-discord\node_modules\bluebird\js\main\async.js:133:10)
ERR! listen     at Immediate.Async.drainQueues (..\Facebook-to-discord\node_modules\bluebird\js\main\async.js:15:14)
ERR! listen     at runCallback (timers.js:789:20)
ERR! listen     at tryOnImmediate (timers.js:751:5)
ERR! listen     at processImmediate [as _immediateCallback] (timers.js:722:5)
ERR! listen  TypeError: config.facebook.client.getThreadInfoGraphQL is not a function
ERR! listen     at module.exports.opts (..\Facebook-to-discord\lib\listeners\facebook.js:19:26)
ERR! listen     at config.facebook.client.listen (..\Facebook-to-discord\index.js:12:51)
ERR! listen     at resolveAttachmentUrl (..\Facebook-to-discord\node_modules\facebook-chat-api\src\listen.js:262:29)
ERR! listen     at parsePackets (..\Facebook-to-discord\node_modules\facebook-chat-api\src\listen.js:281:23)
ERR! listen     at Array.forEach (<anonymous>)
ERR! listen     at ..\Facebook-to-discord\node_modules\facebook-chat-api\src\listen.js:151:14
ERR! listen     at tryCatcher (..\Facebook-to-discord\node_modules\bluebird\js\main\util.js:26:23)
ERR! listen     at Promise._settlePromiseFromHandler (..\Facebook-to-discord\node_modules\bluebird\js\main\promise.js:510:31)
ERR! listen     at Promise._settlePromiseAt (..\Facebook-to-discord\node_modules\bluebird\js\main\promise.js:584:18)
ERR! listen     at Promise._settlePromises (..\Facebook-to-discord\node_modules\bluebird\js\main\promise.js:700:14)
ERR! listen     at Async._drainQueue (..\Facebook-to-discord\node_modules\bluebird\js\main\async.js:123:16)
ERR! listen     at Async._drainQueues (..\Facebook-to-discord\node_modules\bluebird\js\main\async.js:133:10)
ERR! listen     at Immediate.Async.drainQueues (..\Facebook-to-discord\node_modules\bluebird\js\main\async.js:15:14)
ERR! listen     at runCallback (timers.js:789:20)
ERR! listen     at tryOnImmediate (timers.js:751:5)
ERR! listen     at processImmediate [as _immediateCallback] (timers.js:722:5)
Unhandled rejection TypeError: Cannot read property 'threadID' of undefined
    at module.exports.opts (..\Facebook-to-discord\lib\listeners\facebook.js:19:55)
    at config.facebook.client.listen (..\Facebook-to-discord\index.js:12:51)
    at ..\Facebook-to-discord\node_modules\facebook-chat-api\src\listen.js:443:11
    at tryCatcher (..\Facebook-to-discord\node_modules\bluebird\js\main\util.js:26:23)
    at Promise._settlePromiseFromHandler (..\Facebook-to-discord\node_modules\bluebird\js\main\promise.js:510:31)
    at Promise._settlePromiseAt (..\Facebook-to-discord\node_modules\bluebird\js\main\promise.js:584:18)
    at Promise._settlePromises (..\Facebook-to-discord\node_modules\bluebird\js\main\promise.js:700:14)
    at Async._drainQueue (..\Facebook-to-discord\node_modules\bluebird\js\main\async.js:123:16)
    at Async._drainQueues (..\Facebook-to-discord\node_modules\bluebird\js\main\async.js:133:10)
    at Immediate.Async.drainQueues (..\Facebook-to-discord\node_modules\bluebird\js\main\async.js:15:14)
    at runCallback (timers.js:789:20)
    at tryOnImmediate (timers.js:751:5)
    at processImmediate [as _immediateCallback] (timers.js:722:5)
```









## Steps to Reproduce the Problem

  1.  Run the bot with npm start
  2.  Send a message from The FB Side
  3.  Crashes out every time

## Specifications

  - Platform:  Windows 10  Professional
  - Version (can be found in package.json):   2.6.9
  - Node.js version (`node -v`):  v8.9.4

messenger-discord@2.6.9 C:\Users\phil-\Documents\Pokemon Go\miscord-dev\miscord-dev
+-- UNMET PEER DEPENDENCY bufferutil@^3.0.3
+-- chalk@2.3.0
| +-- ansi-styles@3.2.0
| | `-- color-convert@1.9.1
| |   `-- color-name@1.1.3
| +-- escape-string-regexp@1.0.5
| `-- supports-color@4.5.0
|   `-- has-flag@2.0.0
+-- discord.js@11.3.0
| +-- long@3.2.0
| +-- prism-media@0.0.1
| +-- snekfetch@3.6.2
| +-- tweetnacl@1.0.0
| `-- ws@4.0.0
|   +-- async-limiter@1.0.0
|   +-- safe-buffer@5.1.1
|   `-- ultron@1.1.1
+-- emoji-strip@1.0.1
| `-- emoji-regex@6.5.1
+-- UNMET PEER DEPENDENCY erlpack@discordapp/erlpack
+-- facebook-chat-api@1.4.0 (github:Schmavery/facebook-chat-api#cf797d8e0612e71b88f06648f9fccb54d3955ef7)
| +-- bluebird@2.11.0
| +-- cheerio@0.19.0
| | +-- css-select@1.0.0
| | | +-- boolbase@1.0.0
| | | +-- css-what@1.0.0
| | | +-- domutils@1.4.3
| | | | `-- domelementtype@1.3.0 deduped
| | | `-- nth-check@1.0.1
| | |   `-- boolbase@1.0.0 deduped
| | +-- dom-serializer@0.1.0
| | | +-- domelementtype@1.1.3
| | | `-- entities@1.1.1 deduped
| | +-- entities@1.1.1
| | +-- htmlparser2@3.8.3
| | | +-- domelementtype@1.3.0
| | | +-- domhandler@2.3.0
| | | | `-- domelementtype@1.3.0 deduped
| | | +-- domutils@1.5.1
| | | | +-- dom-serializer@0.1.0 deduped
| | | | `-- domelementtype@1.3.0 deduped
| | | +-- entities@1.0.0
| | | `-- readable-stream@1.1.14
| | |   +-- core-util-is@1.0.2 deduped
| | |   +-- inherits@2.0.3 deduped
| | |   +-- isarray@0.0.1
| | |   `-- string_decoder@0.10.31
| | `-- lodash@3.10.1
| +-- npmlog@1.2.1
| | +-- ansi@0.3.1
| | +-- are-we-there-yet@1.0.6
| | | +-- delegates@1.0.0 deduped
| | | `-- readable-stream@1.1.14 deduped
| | `-- gauge@1.2.7
| |   +-- ansi@0.3.1 deduped
| |   +-- has-unicode@2.0.1 deduped
| |   +-- lodash.pad@4.5.1
| |   +-- lodash.padend@4.6.1
| |   `-- lodash.padstart@4.6.1
| +-- prettier@1.11.1
| `-- request@2.85.0
|   +-- aws-sign2@0.7.0
|   +-- aws4@1.6.0
|   +-- caseless@0.12.0
|   +-- combined-stream@1.0.5
|   | `-- delayed-stream@1.0.0
|   +-- extend@3.0.1 deduped
|   +-- forever-agent@0.6.1
|   +-- form-data@2.3.1 deduped
|   +-- har-validator@5.0.3
|   | +-- ajv@5.5.2
|   | | +-- co@4.6.0
|   | | +-- fast-deep-equal@1.1.0
|   | | +-- fast-json-stable-stringify@2.0.0
|   | | `-- json-schema-traverse@0.3.1
|   | `-- har-schema@2.0.0
|   +-- hawk@6.0.2
|   | +-- boom@4.3.1
|   | | `-- hoek@4.2.1 deduped
|   | +-- cryptiles@3.1.2
|   | | `-- boom@5.2.0
|   | |   `-- hoek@4.2.1 deduped
|   | +-- hoek@4.2.1
|   | `-- sntp@2.1.0
|   |   `-- hoek@4.2.1 deduped
|   +-- http-signature@1.2.0
|   | +-- assert-plus@1.0.0
|   | +-- jsprim@1.4.1
|   | | +-- assert-plus@1.0.0 deduped
|   | | +-- extsprintf@1.3.0
|   | | +-- json-schema@0.2.3
|   | | `-- verror@1.10.0
|   | |   +-- assert-plus@1.0.0 deduped
|   | |   +-- core-util-is@1.0.2 deduped
|   | |   `-- extsprintf@1.3.0 deduped
|   | `-- sshpk@1.14.1
|   |   +-- asn1@0.2.3
|   |   +-- assert-plus@1.0.0 deduped
|   |   +-- bcrypt-pbkdf@1.0.1
|   |   | `-- tweetnacl@0.14.5
|   |   +-- dashdash@1.14.1
|   |   | `-- assert-plus@1.0.0 deduped
|   |   +-- ecc-jsbn@0.1.1
|   |   | `-- jsbn@0.1.1 deduped
|   |   +-- getpass@0.1.7
|   |   | `-- assert-plus@1.0.0 deduped
|   |   +-- jsbn@0.1.1
|   |   `-- tweetnacl@0.14.5
|   +-- is-typedarray@1.0.0
|   +-- isstream@0.1.2
|   +-- json-stringify-safe@5.0.1
|   +-- mime-types@2.1.17
|   | `-- mime-db@1.30.0
|   +-- oauth-sign@0.8.2
|   +-- performance-now@2.1.0
|   +-- qs@6.5.1 deduped
|   +-- safe-buffer@5.1.1 deduped
|   +-- stringstream@0.0.5
|   +-- tough-cookie@2.3.4
|   | `-- punycode@1.4.1
|   +-- tunnel-agent@0.6.0
|   | `-- safe-buffer@5.1.1 deduped
|   `-- uuid@3.2.1
+-- UNMET PEER DEPENDENCY libsodium-wrappers@^0.5.4
+-- UNMET PEER DEPENDENCY node-opus@^0.2.7
+-- npmlog@4.1.2
| +-- are-we-there-yet@1.1.4
| | +-- delegates@1.0.0
| | `-- readable-stream@2.3.5
| |   +-- core-util-is@1.0.2 deduped
| |   +-- inherits@2.0.3 deduped
| |   +-- isarray@1.0.0
| |   +-- process-nextick-args@2.0.0 deduped
| |   +-- safe-buffer@5.1.1 deduped
| |   +-- string_decoder@1.0.3
| |   | `-- safe-buffer@5.1.1 deduped
| |   `-- util-deprecate@1.0.2 deduped
| +-- console-control-strings@1.1.0
| +-- gauge@2.7.4
| | +-- aproba@1.2.0
| | +-- console-control-strings@1.1.0 deduped
| | +-- has-unicode@2.0.1
| | +-- object-assign@4.1.1
| | +-- signal-exit@3.0.2
| | +-- string-width@1.0.2
| | | +-- code-point-at@1.1.0
| | | +-- is-fullwidth-code-point@1.0.0
| | | | `-- number-is-nan@1.0.1
| | | `-- strip-ansi@3.0.1 deduped
| | +-- strip-ansi@3.0.1
| | | `-- ansi-regex@2.1.1
| | `-- wide-align@1.1.2
| |   `-- string-width@1.0.2 deduped
| `-- set-blocking@2.0.0
+-- UNMET PEER DEPENDENCY opusscript@^0.0.6
+-- remove-accents@0.4.2
+-- UNMET PEER DEPENDENCY sodium@^2.0.3
+-- superagent@3.8.2
| +-- component-emitter@1.2.1
| +-- cookiejar@2.1.1
| +-- debug@3.1.0
| | `-- ms@2.0.0
| +-- extend@3.0.1
| +-- form-data@2.3.1
| | +-- asynckit@0.4.0
| | +-- combined-stream@1.0.5 deduped
| | `-- mime-types@2.1.17 deduped
| +-- formidable@1.2.0
| +-- methods@1.1.2
| +-- mime@1.6.0
| +-- qs@6.5.1
| `-- readable-stream@2.3.5
|   +-- core-util-is@1.0.2
|   +-- inherits@2.0.3
|   +-- isarray@1.0.0
|   +-- process-nextick-args@2.0.0
|   +-- safe-buffer@5.1.1 deduped
|   +-- string_decoder@1.0.3
|   | `-- safe-buffer@5.1.1 deduped
|   `-- util-deprecate@1.0.2
+-- UNMET PEER DEPENDENCY uws@^9.14.0
`-- yn@2.0.0

npm ERR! peer dep missing: bufferutil@^3.0.3, required by discord.js@11.3.0
npm ERR! peer dep missing: erlpack@discordapp/erlpack, required by discord.js@11.3.0
npm ERR! peer dep missing: node-opus@^0.2.7, required by discord.js@11.3.0
npm ERR! peer dep missing: opusscript@^0.0.6, required by discord.js@11.3.0
npm ERR! peer dep missing: sodium@^2.0.3, required by discord.js@11.3.0
npm ERR! peer dep missing: libsodium-wrappers@^0.5.4, required by discord.js@11.3.0
npm ERR! peer dep missing: uws@^9.14.0, required by discord.js@11.3.0

--- a/lib/discord/getChannel.js
+++ b/lib/discord/getChannel.js
@@ -8,7 +8,7 @@ module.exports = async params => {
   // try searching by topic
   var channel = config.discord.channels.get(topic)
   // if found, change name and return
-  log.silly('getChannel', channel)
+  log.silly('getChannel', 'existing channel:', channel)
   if (channel) return config.discord.renameChannels ? (channel.name === name ? channel : channel.edit({ name })) : channel
 
   // if not found, create new channel with specified name and set its parent and topic
@@ -16,6 +16,6 @@ module.exports = async params => {
   config.discord.channels.set(topic, channel)
   channel.setParent(config.discord.category)
   channel.setTopic(topic)
-  log.silly('getChannel', channel)
+  log.silly('getChannel', 'new channel:', channel)
   return channel
 }

--- a/lib/listeners/facebook.js
+++ b/lib/listeners/facebook.js
@@ -16,7 +16,7 @@ module.exports = opts => {
   log.silly('facebookListener', 'Facebook message: %o', message)
 
   // get thread info to know if it's a group conversation
-  config.facebook.client.getThreadInfoGraphQL(message.threadID, (err, thread) => {
+  config.facebook.client.getThreadInfo(message.threadID, (err, thread) => {
     if (err) return log.error('facebookListener', err)
     log.verbose('facebookListener', 'Got Facebook thread')
     log.silly('facebookListener', 'Thread:', thread)
@@ -32,7 +32,7 @@ module.exports = opts => {
       log.verbose('facebookListener', 'Nickname:', nickname)
 
       // get thread name / user
-      var name = thread.threadType === 'one_to_one' ? (nickname || sender[message.senderID].name) : (thread.threadName || thread.threadID)
+      var name = thread.isGroup ? (thread.threadName || thread.threadID) : (nickname || sender[message.senderID].name)
       log.verbose('facebookListener', 'Channel name (before clean):', name)
 
       // clean name for the needs of discord channel naming

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "messenger-discord",
-  "version": "2.6.9",
+  "version": "2.6.10",
   "description": "Facebook Messenger to Discord bridge",
   "main": "index.js",
   "scripts": {

--- a/package.json
+++ b/package.json
@@ -25,7 +25,7 @@
     "chalk": "^2.3.0",
     "discord.js": "^11.3.0",
     "emoji-strip": "^1.0.1",
-    "facebook-chat-api": "github:schmavery/facebook-chat-api",
+    "facebook-chat-api": "github:Schmavery/facebook-chat-api",
     "npmlog": "^4.1.2",
     "remove-accents": "^0.4.2",
     "superagent": "^3.8.2",

